### PR TITLE
Azure Stack Backup Admin Specification

### PR DESCRIPTION
**Pull request by deathly809**
This is for Azure Stack **ONLY**.  This will not be used in Azure.

### PR information
- [X] The title of the PR is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [X] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [X] If applicable, the PR references the bug/issue that it fixes.
- [X] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [X] My spec meets the review criteria:
  - [X] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [X] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [X] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backup.json [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+                "code": {
+                    "description": "Operation result code.",
+                    "type": "string"
+                },
+                "message": {
+                    "description": "Description of the operation status.",
+                    "type": "string"
+                }
+            }
+        }
```
Comment (anuchandy) [inline specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backup.json]: Can we mark properties of Operation, Display and LongRunningOperationStatus as `readonly` also the properties OperationList.value, OperationList.nextLink?

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backups.json [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+                    "404": {
+                        "description": "NOT FOUND"
+                    }
+                }
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Backup.Admin/backupLocations/{location}/backups/{backup}/restore": {
+            "post": {
+                "x-ms-examples": {
+                    "Get a backup from a location based on name.": {
```
Comment (anuchandy) [inline specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backups.json]: Please correct the description (it's restore backup not get backup, right?)

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+                    "description": "Password to access the location.",
+                    "type": "string"
+                },
+                "encryptionKeyBase64": {
+                    "description": "Encryption key.",
+                    "type": "string"
+                },
+                "backupFrequencyInHours": {
+                    "description": "The interval, in hours, for the frequency that the scheduler takes a backup.",
+                    "type": "string"
```
Comment (anuchandy) [inline specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json]: should this be  an integer like `backupRetentionPeriodInDays`?

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json [hunk 4]
```diff
@@ -0,0 +1,315 @@
...
+                    "description": "True if the backup scheduler is enabled.",
+                    "type": "boolean"
+                },
+                "nextBackupTime": {
+                    "description": "The scheduled time of the next backup.",
+                    "type": "string"
+                },
+                "lastBackupTime": {
+                    "description": "Time of backup.",
+                    "type": "string"
```
Comment (anuchandy) [inline specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json]: Should we use date type here?

```json
        "lastBackupTime": {
          "type": "string",
          "format": "date-time",
          "description": "Time of backup."
        }
```

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json [hunk 5]
```diff
@@ -0,0 +1,315 @@
...
+                    },
+                    "404": {
+                        "description": "NOT FOUND"
+                    }
+                },
+                "x-ms-long-running-operation": true
+            }
+        },
+        "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.Backup.Admin/backupLocations/{location}/createBackup": {
+            "post": {
```
Comment (anuchandy) [inline specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json]: just trying to understand, what exactly `{location}` is? is it Azure regions?

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backups.json [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+                },
+                "stampVersion": {
+                    "description": "Current version.",
+                    "type": "string"
+                },
+                "oemVersion": {
+                    "description": "OEM version.",
+                    "type": "string"
+                }
+            }
```
Comment (anuchandy) [inline specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backups.json]: Is it possible to mark properties  of `BackupInfo` as readonly? i don't see it is used in create or update so thought it is eligible to be marked as readonly

**Review by anuchandy — CHANGES_REQUESTED** [review]

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json (hunk 3 above)
Comment (daozha) [reply specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json]: According to BRP API, this should be integer.

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backup.json (hunk 1 above)
Comment (deathly809) [reply specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backup.json]: Done.

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backups.json (hunk 2 above)
Comment (deathly809) [reply specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backups.json]: Done.

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json (hunk 3 above)
Comment (deathly809) [reply specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json]: Done.

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json (hunk 4 above)
Comment (deathly809) [reply specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/BackupLocations.json]: Leaving as a string.  I have not seen any returned value which I can verify is a date-time.  String is more general and if result is not a date-time we will get a deserialization error.

File: specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backups.json (hunk 6 above)
Comment (deathly809) [reply specification/azsadmin/resource-manager/backup/Microsoft.Backup.Admin/preview/2016-05-01/Backups.json]: Done.

**Comment by deathly809** [comment]
I made the requested changes and validated locally using OAV.

**Comment by anuchandy** [comment]
@deathly809, @ daozha thanks, looks good.

**Review by anuchandy — APPROVED** [review]

**Pull request merged**